src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -0,0 +1,369 @@
+";
+            var compilation = CreatePatternCompilation(source);
+            compilation.VerifyDiagnostics(
+                );
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+
+            var subpatterns = tree.GetRoot().DescendantNodes().OfType<SubpatternSyntax>().ToArray();
+            Assert.Equal(2, subpatterns.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0]));
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0].NameColon));
+            var x = subpatterns[0].NameColon.Name;
+            var xSymbol = model.GetSymbolInfo(x);
+            Assert.Equal(CandidateReason.None, xSymbol.CandidateReason);
+            Assert.Equal("System.Int32 (System.Int32 X, System.Int32 Y).X", xSymbol.Symbol.ToTestDisplayString());
+            Assert.Equal(0, xSymbol.CandidateSymbols.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[1]));
+            AssertEmpty(model.GetSymbolInfo(subpatterns[1].NameColon));
+            var y = subpatterns[1].NameColon.Name;
+            var ySymbol = model.GetSymbolInfo(y);
+            Assert.Equal(CandidateReason.None, ySymbol.CandidateReason);
+            Assert.Equal("System.Int32 (System.Int32 X, System.Int32 Y).Y", ySymbol.Symbol.ToTestDisplayString());
+            Assert.Equal(0, ySymbol.CandidateSymbols.Length);
+        }
+
+        [Fact]
+        public void PositionalPatternSymbolInfo_04()
+        {
+            var source =
+@"
+using System;
+class Program
+{
+    public static void Main()
+    {
+        var p = (Z: 3, W: 4);
+        Console.WriteLine(p is ( X: 3, Y: 4 ));
+    }
+}
+";
+            var compilation = CreatePatternCompilation(source);
+            compilation.VerifyDiagnostics(
+                // (8,34): error CS8416: The name 'X' does not identify tuple element 'Item1'.
+                //         Console.WriteLine(p is ( X: 3, Y: 4 ));
+                Diagnostic(ErrorCode.ERR_TupleElementNameMismatch, "X").WithArguments("X", "Item1").WithLocation(8, 34),
+                // (8,40): error CS8416: The name 'Y' does not identify tuple element 'Item2'.
+                //         Console.WriteLine(p is ( X: 3, Y: 4 ));
+                Diagnostic(ErrorCode.ERR_TupleElementNameMismatch, "Y").WithArguments("Y", "Item2").WithLocation(8, 40)
+                );
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+
+            var subpatterns = tree.GetRoot().DescendantNodes().OfType<SubpatternSyntax>().ToArray();
+            Assert.Equal(2, subpatterns.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0]));
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0].NameColon));
+            var x = subpatterns[0].NameColon.Name;
+            var xSymbol = model.GetSymbolInfo(x);
+            Assert.Equal(CandidateReason.None, xSymbol.CandidateReason);
+            Assert.Null(xSymbol.Symbol);
```

Comment:
>Null [](start = 19, length = 4)

Why is the behavior with a tuple using different names different than the behavior with a Deconstruct using different names (covered by `PositionalPatternSymbolInfo_02`)?

Reply:
In a `Deconstruct`, we can tell what symbol is uniquely identified by the identifier merely by the position. If the name is incorrect, it is and error, but it still refers to that (positional) parameter.

In a tuple, there may be more than one name for a given tuple position (e.g. `x` and `Item1`). So in that case we use name lookup to decide what element field was intended.

---
In reply to: [191020864](https://github.com/dotnet/roslyn/pull/27160#discussion_r191020864) [](ancestors = 191020864)